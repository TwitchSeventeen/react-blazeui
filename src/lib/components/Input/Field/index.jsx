import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function Field(props) {
  const {
    className, iconLeft, iconRight, children, fixed, ...other
  } = props;
  const classes = classNames(
    'o-field',
    { 'o-field--icon-left': iconLeft },
    { 'o-field--icon-right': iconRight },
    { 'o-field--fixed': fixed },
    className,
  );

  return (
    <div {...other} className={classes}>
      { children }
    </div>
  );
}

Field.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
  /**
   * Additional classes
   */
  className: PropTypes.string,
  /**
   * Defines a left icon
   */
  iconLeft: PropTypes.bool,
  /**
   * Defines a right icon
   */
  iconRight: PropTypes.bool,
  /**
   * Defines a fixed with field
   */
  fixed: PropTypes.bool,
};

Field.defaultProps = {
  className: null,
  iconLeft: false,
  iconRight: false,
  children: null,
  fixed: false,
};

export default Field;
