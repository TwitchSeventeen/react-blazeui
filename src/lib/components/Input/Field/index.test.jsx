import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Field from './index';

describe('Field', () => {
  it('should be defined', () => {
    expect(Field).toBeDefined();
  });

  it('renders correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
        <Field>
          lorem ipsum dolor
        </Field>
    );

    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
