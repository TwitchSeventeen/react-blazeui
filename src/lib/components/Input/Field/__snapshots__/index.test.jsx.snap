// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field renders correctly 1`] = `
<div
  className="o-field"
  fixed={false}
  iconLeft={false}
  iconRight={false}
>
  lorem ipsum dolor
</div>
`;
